Use react-fontawesome instead of using fontawesome directly

tacigar opened the pull request:
Before, Zipkin Lens used Font-Awesome directly, but I think this way has the following problems.

* Additional configs are required for Webpack (see: https://github.com/openzipkin/zipkin/blob/08cee5023bf6e24c277bce85839ad1c043230f63/zipkin-lens/webpack.dev.config.js#L51-L54)
* All icons need to be imported, so bundle size will increase (see: https://github.com/openzipkin/zipkin/blob/08cee5023bf6e24c277bce85839ad1c043230f63/zipkin-lens/scss/main.scss#L19-L22)

I think it is better to use Font-Awesome through React components provided by Fort-Awesome (https://github.com/FortAwesome/react-fontawesome).
This library seems to avoid such problems.

tacigar committed:
Remove @fortawesome/fontawesome-free

tacigar committed:
Add packages

tacigar committed:
Remove importing fontawesome statements from main.scss

tacigar committed:
Refactor Sidebar and use icon components of react-fontawesome

tacigar committed:
Remove nonsense unit tests

tacigar committed:
Refactor App components

tacigar committed:
Use react-fontawesome to TraceJsonUploader

tacigar committed:
Use react-fontawesome to GlobalSearch

tacigar committed:
Use react-fontawesome to ServiceBadge

tacigar committed:
Use react-fontawesome to DependenciesPage

tacigar committed:
Use react-fontawesome to DiscoverPage

tacigar committed:
Use react-fontawesome to TracePage

tacigar committed:
Remove unnecessary import statements

tacigar committed:
Update webpack configs

anuraaga reviewed (commented):
Yes!! Thanks a lot for this. I wasn't expecting big styling changes for this though. Is it possible to just migrate libraries with less code change? If not no worries will take a deeper look

tacigar commented:
Sorry for big changes.
In addition to library migration, I changed the Sidebar code a bit.
But all other changes are library migrations.

The SidebarMenuItem component received a FontAwesome icon string as props.

https://github.com/openzipkin/zipkin/blob/08cee5023bf6e24c277bce85839ad1c043230f63/zipkin-lens/src/components/App/Sidebar.jsx#L63-L66

So it was necessary to change SidebarMenuItem's props when migrating the library.
When making that change, I realized that SidebarMenuIcon has too verbose and meaningless logic.
So I refactored the Sidebar code a bit.

If possible, I would appreciate it if you could review this change within this PR... :sob: 
Sorry...

tacigar commented on zipkin-lens/src/components/App/Layout.jsx:
```diff
@@ -13,33 +13,90 @@
-      overflow="auto"
-    >
-      {children}
+const Layout = ({ children }) => {
```
I have integrated Sidebar component into Layout component, since they are very simple components,

tacigar commented on zipkin-lens/src/components/App/Layout.jsx:
```diff
@@ -13,33 +13,90 @@
+
+const drawerWidth = '3.2rem';
+
+const useStyles = makeStyles(theme => ({
```
This style is almost the same as the previous one.

tacigar replied:
This combines the styles of the previous Sidebar component and the previous Layout component.

tacigar commented on zipkin-lens/src/components/App/Layout.jsx:
```diff
@@ -13,33 +13,90 @@
+  return (
+    <Box className={classes.root}>
+      <CssBaseline />
+      <Drawer open variant="permanent" className={classes.drawer} classes={{ paper: classes.drawerPaper }}>
```
This Drawer was previously in Sidebar component.

tacigar commented on zipkin-lens/src/components/App/Layout.test.js:
```diff
@@ -1,62 +0,0 @@
-/*
```
I thought these unit tests were meaningless and would prevent us from changing structures of the DOM tree in the future.
So I deleted these unit tests.

abesto replied:
Was gonna ask about this, thanks for explaining your reasoning.

tacigar commented on zipkin-lens/src/components/App/SidebarMenu.jsx:
```diff
@@ -0,0 +1,76 @@
+  },
+}));
+
+export const SidebarMenuImpl = ({
```
I renamed SidebarMenuItem to SidebarMenu.

anuraaga reviewed (approved):
Thanks!

abesto reviewed (approved):
+1 for more clearly separated changes, but the individual commits in here are actually doing an OK job of that I think. Methinks this is a good change 👍
